api/src/main/java/org/apache/iceberg/transforms/Bucket.java

```diff
@@ -41,7 +41,7 @@
 import static org.apache.iceberg.types.Type.TypeID;
 
 abstract class Bucket<T> implements Transform<T, Integer> {
-  private static final HashFunction MURMUR3 = Hashing.murmur3_32();
+  private static final HashFunction MURMUR3 = Hashing.murmur3_32_fixed();
```

Comment:
We should have hash stability tests to ensure hash values don't change. See https://github.com/trinodb/trino/blob/863da2b2d3f92dd858e028d30ceb63184dfb515d/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergBucketing.java#L122 for example.

This can be followed up upon, for now we can rely on @wendigo's test in Trino.

Reply:
Tests are in `TestBucketing`: https://github.com/apache/iceberg/blob/master/api/src/test/java/org/apache/iceberg/transforms/TestBucketing.java

This tests all of the examples from the spec, as well as type promotion and some edge cases like UTF-8 with surrogate pairs.

Reply:
@findepi I was just going through the trino code and noticed it uses a different [Murmur3 Implementation](https://github.com/trinodb/trino/blob/fb88c0bf589db55d77223fc31aab29acbb2d0698/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTransforms.java#L422), does this match the fixed version here?

Reply:
@RussellSpitzer yes, we use different murmur3 implementation, that's how https://github.com/google/guava/issues/5648 was found.
Yes, it matches the fixed version.